Modules/UserRoles.cs

```diff
@@ -0,0 +1,157 @@
+        }
+
+        public static async Task RemoveUserRolesAsync(CommandContext ctx, Func<System.Reflection.PropertyInfo, bool> predicate)
+        {
+            if (Program.cfgjson.UserRoles is null)
+            {
+                // Config hasn't been updated yet.
+                return;
+            }
+
+            DiscordGuild guild = await Program.discord.GetGuildAsync(ctx.Guild.Id);
+            System.Reflection.PropertyInfo[] roleIds = Program.cfgjson.UserRoles.GetType().GetProperties().Where(predicate).ToArray();
+            foreach (System.Reflection.PropertyInfo roleId in roleIds)
+            {
+                DiscordRole roleToGrant = guild.GetRole((ulong)roleId.GetValue(Program.cfgjson.UserRoles, null));
+                await ctx.Member.RevokeRoleAsync(roleToGrant);
+            }
+
+            await ctx.Message.DeleteAsync();
+        }
+
+    }
+    [UserRolesPresent]
+    public class UserRoleCmds : BaseCommandModule
+    {
+        [
+            Command("join-insider-dev"),
+            Description("Gives you the Windows Insiders (Dev) role"),
+            HomeServer
+        ]
+        public async Task JoinInsiderDevCmd(CommandContext ctx)
+        {
+            await UserRoles.GiveUserRoleAsync(ctx, Program.cfgjson.UserRoles.InsiderDev);
+        }
+
+        [
+            Command("join-insider-beta"),
+            Description("Gives you the Windows Insiders (Beta) role"),
+            HomeServer
+        ]
+        public async Task JoinInsiderBetaCmd(CommandContext ctx)
+        {
+            await UserRoles.GiveUserRoleAsync(ctx, Program.cfgjson.UserRoles.InsiderBeta);
+        }
+
+        [
+            Command("join-insider-rp"),
+            Description("Gives you the Windows Insiders (Release Preview) role"),
+            HomeServer
+        ]
+        public async Task JoinInsiderRPCmd(CommandContext ctx)
+        {
+            await UserRoles.GiveUserRoleAsync(ctx, Program.cfgjson.UserRoles.InsiderRP);
+        }
+
+        [
+            Command("join-patch-tuesday"),
+            Description("Gives you the 💻 Patch Tuesday role"),
+            HomeServer
+        ]
+        public async Task JoinPatchTuesday(CommandContext ctx)
+        {
+            await UserRoles.GiveUserRoleAsync(ctx, Program.cfgjson.UserRoles.PatchTuesday);
+        }
+
+        [
+            Command("keep-me-updated"),
+            Description("Gives you all opt-in roles"),
+            HomeServer
+        ]
+        public async Task KeepMeUpdated(CommandContext ctx)
+        {
+            await UserRoles.GiveUserRolesAsync(ctx, x => true);
+        }
+
+        [
+            Command("leave-insiders"),
+            Description("Removes you from Insider roles"),
+            Aliases("leave-insider"),
+            HomeServer
+        ]
+        public async Task LeaveInsiders(CommandContext ctx)
+        {
+            await UserRoles.RemoveUserRolesAsync(ctx, x => x.Name.Contains("Insider"));
```

Comment:
This isn't the most elegant way of handling this, I would recommend iterating through the list of IDs in the config.